plugins/velocity/src/main/java/org/apache/struts2/views/velocity/components/AbstractDirective.java

```diff
@@ -57,8 +59,11 @@ public int getType() {
     protected abstract Component getBean(ValueStack stack, HttpServletRequest req, HttpServletResponse res);
 
     public boolean render(InternalContextAdapter ctx, Writer writer, Node node) throws IOException, ResourceNotFoundException, ParseErrorException, MethodInvocationException {
-        // get the bean
-        ValueStack stack = (ValueStack) ctx.get("stack");
+        ValueStack stack = extractValueStack(ctx);
+        if (stack == null) {
```

Comment:
This shouldn't occur but left it for backwards compatibility in case there is a scenario I didn't foresee